### PR TITLE
DBZ-356 Close leaked JDBC connection

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-356

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Good catch, @jpechane! There seem to be some more problematic occurances, e.g. in `RecordsStreamProducer#columnValues()`. Could you check for usages of that method globally? Thanks!

**Comment by gunnarmorling** [comment]
Hey @jpechane, I've pushed one more commit to this PR. Can you review it and rebase + merge if you approve? Thanks!

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java [hunk 1]
```diff
@@ -146,24 +147,50 @@ protected synchronized void commit()  {
...
             }
-        } catch (Exception e) {
-            throw new ConnectException(e.getCause() != null ? e.getCause() : e);
-        } finally {
-            replicationStream.set(null);
-            executorService.shutdownNow();
-            previousContext.restore();
+        }
+        catch(Exception e) {
+            closingException = e;
```
Comment (jpechane) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java]: Nice defensive programming, a single question 
Is not `closingException` lost when `replicationConnection.close()` throws an Exception but `typeResolverConnection` is either `null` or its closing does not throw an exception?

**Comment by gunnarmorling** [comment]
Agh, you are right. I've pushed one more commit which should take care of
it. This entire code is quite convoluted, I'm wondering whether there isn't
a better way to do that...

**Comment by jpechane** [comment]
Good, thanks a lot. Regarding better way I think it would need lambdas as a template. Not necessary to use it now.

**Pull request merged**